lib/takeaway.rb

```diff
@@ -0,0 +1,31 @@
+require_relative 'menu'
+require_relative 'confirmation'
+require_relative 'text'
+
+
+class Takeaway
+
+	attr_reader :menu, :checkout
+
+	def initialize menu
+		@menu = menu
+		@checkout = {}
+	end
+	
+	def show
+		menu.each { |meal,price| print "#{meal}		 £#{price}\n"}
+	end
+
+	def place_order(dish, qty)
+		 @checkout[dish] = [menu[dish],qty] 
+	end
+
+	def confirm(user_total, confirmation = Confirmation.new(checkout))
+		confirmation.user_cost(user_total)
+	end
```

Comment:
Not sure if this needs a separate class - I am in two minds. On the 1 hand it does help to simplify this class, but on the other hand I'm not sure I see a clear responsibility (the confirmation could be handled by a `fail` at the point of confirmation.). At risk of over-engineering.